tools/skbstat_example.txt

```diff
@@ -0,0 +1,103 @@
+Demonstrations of skbstat, the Linux BPF/bcc version.
+
+
+skbstat prints detailed information about the packet (skb) processed by the kernel,
+including input/output/consume/drop. we can pass filter parameters to it, such as "-p tcp -P 80":
+
+# skbstat.py -p tcp -P 80 --input --output
+TIME     NUM    FUNC/STACK
+17:53:34        [INGRESS]
+         5      tcp_v4_rcv
+         5      ip_local_deliver_finish
+         5      ip_local_deliver
+         1      ip_route_input_noref
+         5      ip_rcv_finish
+         5      ip_rcv
+
+TIME     NUM    FUNC/STACK
+17:53:34        [EGRESS]
+         7      ip_output
+         7      ip_finish_output
+         7      ip_finish_output2
+         7      net:net_dev_queue
+         7      net:net_dev_xmit
+
+TIME     NUM    FUNC/STACK
+17:53:34        [DROP]
+         1
+                kfree_skb+0x1
+                sk_stream_kill_queues+0x52
+                inet_csk_destroy_sock+0x4f
+                tcp_fin+0xe7
+                tcp_data_queue+0x572
+                tcp_rcv_state_process+0x28d
+                tcp_v4_do_rcv+0x70
+                tcp_v4_rcv+0x91b
+                ip_protocol_deliver_rcu+0x36
+                ip_local_deliver_finish+0x44
+                ip_local_deliver+0x6b
+                ip_rcv+0x52
+                __netif_receive_skb_one_core+0x50
+                netif_receive_skb_internal+0x2f
+                napi_gro_receive+0x107
+
+[...]
+
+The "NUM" column shows the number of function calls or skb dropped stacks.
+The "FUNC/STACK" column shows either the function name or function backtrace.
+
+This tool is useful for debugging why a particular packet (skb) was dropped.
+
+For example, when you see HTTP requsts arrive at your machine by tcpdump,
+and get lost in kernel. Then you can run `skbstat -p tcp -P 80` to see which
+kernel routine dropped those packets.
+
+WARNING: When run with `--input/output/consume` options,
+this tool will trace high traffic kernel functions and may lead to
```

Comment:
The example file is used to show examples. Let us remove the WARNING here.

Reply:
ok